### PR TITLE
[iOS][notifications] feature: customize notification preview placeholders

### PR DESCRIPTION
**Pull request by cruzach**
# Why

Allows users to customize the placeholder used on iOS for notifications when previews are disabled in the app's settings (otherwise this defaults to `Notification`)

# How

followed apple's [docs](https://developer.apple.com/documentation/usernotifications/unnotificationcategory/2873733-categorywithidentifier?language=objc)

# Test Plan

Tested using `presentLocalNotification`

### CODE REVIEW

**Review by sjchmiela — APPROVED** [review]
LGTM! 🎉

File: ios/Exponent/Versioned/Core/Api/EXNotifications.m [hunk 1]
```diff
@@ -306,6 +307,7 @@ - (instancetype)initWithExperienceId:(NSString *)experienceId kernelServiceDeleg
   UNNotificationCategory *newCategory = [UNNotificationCategory categoryWithIdentifier:[self internalIdForIdentifier:categoryId]
                                                                                actions:actionsArray
                                                                      intentIdentifiers:@[]
+                                                                     hiddenPreviewsBodyPlaceholder: previewPlaceholder
```
Comment (sjchmiela) [inline ios/Exponent/Versioned/Core/Api/EXNotifications.m]: Actually, I dug in to the Apple documentation (following my gut feeling) and it looks like this initializer (`+categoryWithIdentifier:actions:intentIdentifiers:hiddenPreviewsBodyPlaceholder:options:`) [has been introduced in iOS 11](https://developer.apple.com/documentation/usernotifications/unnotificationcategory/2873733-categorywithidentifier?language=objc), which means it won't be available on iOS 10 (on which the current one is used).

I think the solution would be to `if @available` the initializer and choose the proper one accordingly. Xcode should suggest a quick-fix solution. 🙂

**Review by sjchmiela — CHANGES_REQUESTED** [review]
Just one comment, apart from that—👍 !

File: docs/pages/versions/unversioned/sdk/notifications.md [hunk 2]
```diff
@@ -161,6 +161,7 @@ Registers a new set of actions under given `name`.
   - **isDestructive (_boolean_)** -- (iOS only) If this property is truthy, on iOS the button title will be highlighted (as if [this native option](https://developer.apple.com/documentation/usernotifications/unnotificationactionoptions/1648199-destructive) was set)
   - **isAuthenticationRequired (_boolean_)** -- (iOS only) If this property is truthy, triggering the action will require authentication from the user (as if [this native option](https://developer.apple.com/documentation/usernotifications/unnotificationactionoptions/1648196-authenticationrequired) was set)
   - **doNotOpenInForeground (_boolean_)** -- (iOS only) If this property is truthy, triggering the action will not open the app in foreground (as if [this native option](https://developer.apple.com/documentation/usernotifications/unnotificationactionoptions/unnotificationactionoptionforeground) was **NOT** set)
+- **previewPlaceholder (_string_)** -- (iOS only) Customizable placeholder for the notification preview text. This is shown if the user has disabled notification previews for the app. Defaults to `Notification`.
```
Comment (ide) [inline docs/pages/versions/unversioned/sdk/notifications.md]:
```suggestion
- **previewPlaceholder (_string_)** -- (iOS only) Customizable placeholder for the notification preview text. This is shown if the user has disabled notification previews for the app. Defaults to the localized iOS system default placeholde r(`Notification`).
```

File: ios/Exponent/Versioned/Core/Api/EXNotifications.m (hunk 1 above)
Comment (ide) [inline ios/Exponent/Versioned/Core/Api/EXNotifications.m]: If we pass in nil, does this default to the system default?

Comment (cruzach) [reply ios/Exponent/Versioned/Core/Api/EXNotifications.m]: yeah, just tested that

Comment (cruzach) [reply ios/Exponent/Versioned/Core/Api/EXNotifications.m]: absolutely right, thanks for pointing that out 🙏 fixed it

**Review by ide — APPROVED** [review]

**Pull request merged**